src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/CancellationTokenParametersMustComeLast.cs

```diff
@@ -37,76 +37,88 @@ public override void Initialize(AnalysisContext context)
+                    }
 
-                        // Skip optional parameters, UNLESS one of them is a CancellationToken
-                        // AND it's not the last one.
-                        if (last >= 0 && methodSymbol.Parameters[last].IsOptional
-                            && !methodSymbol.Parameters[last].Type.Equals(cancellationTokenType))
-                        {
-                            last--;
+                    // Skip optional parameters, UNLESS one of them is a CancellationToken
+                    // AND it's not the last one.
+                    if (last >= 0 && methodSymbol.Parameters[last].IsOptional
+                        && !methodSymbol.Parameters[last].Type.Equals(cancellationTokenType))
+                    {
+                        last--;
 
-                            while (last >= 0 && methodSymbol.Parameters[last].IsOptional)
+                        while (last >= 0 && methodSymbol.Parameters[last].IsOptional)
+                        {
+                            if (methodSymbol.Parameters[last].Type.Equals(cancellationTokenType))
                             {
-                                if (methodSymbol.Parameters[last].Type.Equals(cancellationTokenType))
-                                {
-                                    symbolContext.ReportDiagnostic(Diagnostic.Create(
-                                        Rule, methodSymbol.Locations.First(), methodSymbol.ToDisplayString()));
-                                }
-
-                                last--;
+                                symbolContext.ReportDiagnostic(Diagnostic.Create(
+                                    Rule, methodSymbol.Locations.First(), methodSymbol.ToDisplayString()));
                             }
-                        }
 
-                        // Ignore multiple cancellation token parameters at the end of the parameter list.
-                        while (last >= 0 && methodSymbol.Parameters[last].Type.Equals(cancellationTokenType))
-                        {
                             last--;
                         }
+                    }
 
-                        // Ignore parameters passed by reference when they appear at the end of the parameter list.
-                        while (last >= 0 && methodSymbol.Parameters[last].RefKind != RefKind.None)
+                    // Ignore multiple cancellation token parameters at the end of the parameter list.
+                    while (last >= 0 && methodSymbol.Parameters[last].Type.Equals(cancellationTokenType))
+                    {
+                        last--;
+                    }
+
+                    // Ignore parameters passed by reference when they appear at the end of the parameter list.
+                    while (last >= 0 && methodSymbol.Parameters[last].RefKind != RefKind.None)
+                    {
+                        last--;
+                    }
+
+                    // Ignore IProgress<T> when last
```

Comment:
Shall I handle multiple `IProgress` arguments?

Reply:
I've never seen guidelines allowing for this, so I would say only allow one for now. A test would be good so show the current behavior.